### PR TITLE
Adding sql_event_id as an option for sql queries.

### PR DESCRIPTION
**Pull request by hyperbolic2346**
This allows for creating a table of events and a table for files. Start of event generates a new event id with common data and then each file gets an event id. This is different from motion's event number since it is unique to this event across runs. This helps a front-end website match things up properly for display.

Note that I don't have postgresql support for this patch. I'm not sure how postgresql handles keys and things as I do not use it.

### CODE REVIEW

File: event.c [hunk 1]
```diff
@@ -152,106 +152,139 @@ static void on_motion_detected_command(struct context *cnt,
...
-        if ((!strcmp(cnt->conf.database_type, "sqlite3")) && (cnt->conf.database_dbname)) {
-            int res;
-            char *errmsg = 0;
-            res = sqlite3_exec(cnt->database_sqlite3, sqlquery, NULL, 0, &errmsg);
-            if (res != SQLITE_OK ) {
-                MOTION_LOG(ERR, TYPE_DB, NO_ERRNO, "%s: SQLite error was %s",
-                           errmsg);
-                sqlite3_free(errmsg);
-            }
+    if ((!strcmp(cnt->conf.database_type, "sqlite3")) && (cnt->conf.sqlite3_db)) {
```
Comment (Mr-Dave) [inline event.c]: This looks like it was applied against a older version of the motion code.  The sqlite3_db configuration variable was eliminated and replaced with the more generic database_dbname variable (See code above this line that is being replaced)

**Comment by Mr-Dave** [comment]
Just to be clear on the intent.  The event id added here is completely unique.  Like a date/time/microsecond type of ID and the event IDs within motion always start with 1 every time you start it up.  

Next, I don't use the postgresql either but I see changes in the code that are being applied to it.  I think we'd want to leave that as unchanged as possible until it is fully understood.

File: event.c (hunk 1 above)
Comment (hyperbolic2346) [inline event.c]: This is correct, I've been dragging this code along with me for a long time now. I'll get this fixed up.

**Comment by hyperbolic2346** [comment]
Yes, I'm using MySQL to generate unique id's via auto_increment that I can use to tag an event. Then the files spit out by the event can be easily grouped together instead of trying to figure it out from the timestamps.

Postgresql is probably the biggest reason that I haven't pushed this before. I kept meaning on fixing it, but that would mean setting up a new db and I've just been lazy. I can easily just pull any postgresql content, but I have a feeling the support would be exactly the same so I was actually hoping someone with some context could fix it quickly. I'll just pull it for now though.

**Comment by mterzo** [comment]
I ran into this same problem.  I ended up taking the approach that worked better for my environment.  I'm using a redis with a pub/sub and on new videos being written out those send an event to the redis server.  A special bit of code reads the event does some post processing and then creates database entries.

This could also be solved by using a 2 key system.

Every time motion starts up it creates a for lack of better, time_t which is pretty unique for that start.  That + event_id will be unique always.  If motion is respawning faster than a second you have different problem.

**Comment by tosiara** [comment]
Is there any progress on this pull?

**Comment by hyperbolic2346** [comment]
Sorry, but not yet. I need to do this.

**Comment by tosiara** [comment]
Ok, if you need any assistance, let us know and keep us informed

**Comment by Mr-Dave** [comment]
I think I'm going to want to make some revisions to this before we merge.  The things I noticed were:
1.  We probably want to maintain the existing variable for sql query so as to not break older configuration files.
2.  I'd like to see about storing the database event id in the context rather than passing it via another parameter into the mystrformat.  It was on my mental list to consider moving some of the variables that are already there into the context.
3.  The variable "e" seems to already be taken by the usual C format specs so we'd need to use the new long text version of <code>%{ }</code>

As a database user of Motion, I do have one question.  Why we should even define default values in the motion.h.  Motion supports three different database types which I'd have to guess would need different SQL.  Shouldn't the default be empty strings and users who set up databases should be expected to specify correct values?

**Comment by hyperbolic2346** [comment]
I would be careful with expecting users to know how to form a valid SQL query. I'm not saying that they can't, but I know based on some of the emails in the mailing list that I wouldn't expect it from everyone. Also, it is nice even if you know SQL syntax to have a starting point or a best practices kind of thing. If we just leave it to the reader we could inadvertently encourage some poor database layouts.

**Pull request closed without merging**